src/v/archival/ntp_archiver_service.cc

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "archival/ntp_archiver_service.h"
+
+#include "archival/logger.h"
+#include "archival/utils.h"
+#include "model/metadata.h"
+#include "s3/client.h"
+#include "s3/error.h"
+#include "storage/disk_log_impl.h"
+#include "storage/fs_utils.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/loop.hh>
+#include <seastar/core/shared_ptr.hh>
+
+#include <fmt/format.h>
+
+#include <stdexcept>
+
+namespace archival {
+
+class arch_policy_iface {
+public:
+    virtual std::optional<manifest>
+    make_local_manifest(storage::log_manager& lm) = 0;
+};
+
+class arch_policy_non_compacted final : public arch_policy_iface {
+public:
+    arch_policy_non_compacted(model::ntp ntp, model::revision_id rev)
+      : _ntp(ntp)
+      , _rev(rev) {}
+
+    std::optional<manifest>
+    make_local_manifest(storage::log_manager& lm) final {
+        manifest tmp(_ntp, _rev);
+        std::optional<storage::log> log = lm.get(_ntp);
+        if (!log) {
+            return std::nullopt;
+        }
+        auto plog = dynamic_cast<storage::disk_log_impl*>(log->get_impl());
```

Comment:
nice, i haven't seen this in a long time. I think we should push this to an implementation file inside a `archival::details..` namespace.